board/safety/safety_hyundai.h

```diff
@@ -165,38 +170,48 @@ static int hyundai_rx_hook(CANPacket_t *to_push) {
       update_sample(&torque_driver, torque_driver_new);
     }
 
-    if (hyundai_longitudinal) {
-      // ACC steering wheel buttons
-      if (addr == 1265) {
-        int button = GET_BYTE(to_push, 0) & 0x7U;
+    // ACC steering wheel buttons
+    if (addr == 1265) {
+      int cruise_button = GET_BYTE(to_push, 0) & 0x7U;
+      int main_button = GET_BIT(to_push, 3U);
+
+      if ((cruise_button == HYUNDAI_BTN_RESUME) || (cruise_button == HYUNDAI_BTN_SET) || (cruise_button == HYUNDAI_BTN_CANCEL) || (main_button != 0)) {
+        last_button_interaction = 0U;
+      } else {
+        last_button_interaction = MIN(last_button_interaction + 1U, PREV_BUTTON_SAMPLES);
```

Comment:
Do we care about using the smallest unsigned int and then clip like this for any optimization reasons?